src/web/Image.tsx

```diff
@@ -250,14 +252,15 @@ export class Image extends React.Component<Types.ImageProps, ImageState> {
             return;
         }
 
-        this.setState({
-            displayUrl: URL.createObjectURL(blob)
-        });
-
         // Save the newly fetched xhr blob url in the cache.
         XhrBlobUrlCache.insert(this.props.source, this.state.displayUrl);
```

Comment:
I think this change breaks the XhrBlobUrlCache. I suspect the cache was relying on synchronous setStates, but at this point in code this.state.displayUrl will be undefined.  I think you need to create the objectURL above , stuff it into cache then call setState

It's also possible that the cache behaviour is already busted in some cases